File: Julia/algorithmic_toolbox/03-small-fibonacci-number.jl
function fibonacci(n)
    k = 1
    a, b = 0, 1
    while k <= n
        a, b = b, a + b
        k += 1
    end
    a
end


function main()
    n = parse(Int, readline())
    result = fibonacci(n)
    println(result)
end


main()

File: Rust/algorithmic_toolbox/src/bin/03_small_fibonacci_number.rs
use std::io;

fn read_int() -> i32 {
    let mut x = String::new();
    io::stdin().read_line(&mut x).unwrap();
    x.trim().parse().unwrap()
}

fn fibonacci(n: i32) -> i32 {
    let mut k = 1;
    let mut a = 0;
    let mut b = 1;
    while k <= n {
        (a, b) = (b, a + b);
        k += 1;
    }
    a
}

fn main() {
    let n = read_int();
    let result = fibonacci(n);
    println!("{}", result);
}
